### PR TITLE
sys/ztimer: add ztimer_mutex_lock_timeout()

### PR DESCRIPTION
**Pull request by maribu**
### Contribution description

As the title says. Additionally, a test application is provided and a `ztimer_xtimer_compat` is extended to also provide `xtimer_mutex_lock_timeout()` via `ztimer_mutex_lock_timeout()`.

### Testing procedure

- `make -C tests/ztimer_mutex_lock_timeout flash test`
- `USEMODULE=ztimer_xtimer_compat make -C tests/xtimer_mutex_lock_timeout flash test`

### Issues/PRs references

None

### CODE REVIEW

File: sys/include/ztimer/xtimer_compat.h [hunk 1]
```diff
@@ -160,6 +160,15 @@ static inline void xtimer_set_wakeup(xtimer_t *timer, uint32_t offset,
     ztimer_set_wakeup(ZTIMER_USEC, timer, offset, pid);
 }
 
+static inline int xtimer_mutex_lock_timeout(mutex_t *mutex, uint64_t us)
+{
+    assert(us <= UINT32_MAX);
+    if (ztimer_mutex_lock_timeout(ZTIMER_USEC, mutex, (uint32_t)us)) {
```
Comment (benpicco) [inline sys/include/ztimer/xtimer_compat.h:166]: not simply 

```suggestion
    return ztimer_mutex_lock_timeout(ZTIMER_USEC, mutex, (uint32_t)us);
```
?

**Comment by benpicco** [comment]
Only `run_test/tests/pkg_libhydrogen/samr21-xpro:llvm` failed again

File: sys/include/ztimer/xtimer_compat.h (hunk 1 above)
Comment (maribu) [reply sys/include/ztimer/xtimer_compat.h:166]: Sadly some impedance matching is needed. I'll add a comment.

**Review by benpicco — APPROVED** [review]

**Comment by maribu** [comment]
Thanks :-)

**Pull request merged**